lib/identity_cache/query_api.rb

```diff
@@ -156,7 +156,13 @@ def set_embedded_association(record, association_name, coder_or_array) #:nodoc:
 
       def get_embedded_association(record, association, options) #:nodoc:
         embedded_variable = record.public_send(options.fetch(:cached_accessor_name))
-        if record.class.reflect_on_association(association).collection?
+        cached_association = if IdentityCache.fetch_returns_relation 
+          record.class.reflect_on_association(association).collection?
+        else
+          embedded_variable.is_a?(Array)
+        end
+
+        if cached_association
```

Comment:
What you really care about is if embedded_variable is array like, which can be checked for with just `if embedded_variable.respond_to?(:to_ary)`, then you can use `embedded_variable.to_ary.map {`... on the following line